components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,3033 +1,1829 @@
+                               surfric, obklen, ptend, ast, cam_in%landfrac, sgh30, pbuf )
+  call physics_update(state, ptend, ztodt, tend)
+  call t_stopf ('vertical_diffusion_tend')
+  
+  obklen(:) = 0.
+  surfric(:) = 0.
 
+  !-----------------------------------------------------------------------------
+  ! Rayleigh friction calculation
+  !-----------------------------------------------------------------------------
+  if (l_rayleigh) then
 
-if (l_rayleigh) then
-    !===================================================
-    ! Rayleigh friction calculation
-    !===================================================
     call t_startf('rayleigh_friction')
     call rayleigh_friction_tend( ztodt, state, ptend)
     call physics_update(state, ptend, ztodt, tend)
     call t_stopf('rayleigh_friction')
 
-    if (do_clubb_sgs) then
-      call check_energy_chng(state, tend, "vdiff", nstep, ztodt, zero, zero, zero, zero)
-    else
-#ifdef MAML
-      call check_energy_chng(state, tend, "vdiff", nstep, ztodt, cam_in%cflx(:,1), zero, &
-           zero, shfavg_in)
-#else
-      call check_energy_chng(state, tend, "vdiff", nstep, ztodt, cam_in%cflx(:,1), zero, &
-          zero, cam_in%shf)
-#endif
-    endif
-    
+    call check_energy_chng(state, tend, "vdiff", nstep, ztodt, &
+                           cam_in%cflx(:,1), zero, zero, cam_in%shf)
     call check_tracers_chng(state, tracerint, "vdiff", nstep, ztodt, cam_in%cflx)
 
-end if ! l_rayleigh
+  end if ! l_rayleigh
 
-if (l_tracer_aero) then
+  !-----------------------------------------------------------------------------
+  !  aerosol dry deposition processes
+  !-----------------------------------------------------------------------------
+  if (l_tracer_aero) then
 
-    !  aerosol dry deposition processes
     call t_startf('aero_drydep')
     call aero_model_drydep( state, pbuf, obklen, surfric, cam_in, ztodt, cam_out, ptend )
     call physics_update(state, ptend, ztodt, tend)
     call t_stopf('aero_drydep')
 
-    !---------------------------------------------------------------------------------
-    !  ... enforce charge neutrality
-    !---------------------------------------------------------------------------------
+    ! enforce charge neutrality
     call charge_fix( ncol, state%q(:,:,:) )
 
-end if ! l_tracer_aero
+  end if ! l_tracer_aero
 
-if (l_gw_drag) then
-    !===================================================
-    ! Gravity wave drag
-    !===================================================
-    call t_startf('gw_tend')
+  !-----------------------------------------------------------------------------
+  ! Gravity wave drag
+  !-----------------------------------------------------------------------------
+  if (l_gw_drag) then
 
+    call t_startf('gw_tend')
     call gw_tend(state, sgh, pbuf, ztodt, ptend, cam_in)
-
     call physics_update(state, ptend, ztodt, tend)
+    call t_stopf('gw_tend')
```

Comment:
shouldn't this timer need to appear after check_energy_chng?

Reply:
It seems like the most common convention in the other physpkg was to leave check_energy_chng outside of the timer calls. I don't really understand the consequences one way or another. It would make sense to put a seperate timer around  all the check_energy_chng calls to accumulate all the time spent there, but that's not done in the non-MMF physpkg, so I didnt do it here. and I tried to be consistent in leaving check_energy_chng outside of the timers, but I havent gone through to check them all.